lib/ok.ex

```diff
@@ -64,9 +64,72 @@ defmodule OK do
   """
   def failure(reason), do: {:error, reason}
 
+  @doc """
```

Comment:
I agree a longer example would be great. However I would really like one that uses the standard library. To that end I have been looking for all the functions in the standard library that follow this pattern. I you can think of a good longer example that would work I would happily add it.